src/osdc/Journaler.cc

```diff
@@ -839,6 +840,12 @@ void Journaler::_finish_read(int r, uint64_t offset, bufferlist& bl)
   assert(r>=0);
 
   ldout(cct, 10) << "_finish_read got " << offset << "~" << bl.length() << dendl;
+  if (bl.length() < length) {
+    ldout(cct, 0) << "_finish_read got less than expected (" << length << ")" << dendl;
+    error = -EINVAL;
```

Comment:
I think we might need to trigger on_readable here, as is done in the preceding error handling